### ISSUE TITLE
Access from inside a K8S cluster to external database via Internet?

### ISSUE THREAD

**Issue by dstroot** [kubernetes/kubernetes#29381]
This is a cross post from Minikube [352](https://github.com/kubernetes/minikube/issues/352) - sorry, wanted to see if this is a Kubernetes hurdle or a Minikube hurdle:   

Minikube sets up a local network on your host box so your dev box and minikube can talk to each other. If I run a container inside minikube that needs access to resources/services **on the internet** (not just the local machine) is there anything special that needs to be done? Or should it "just work"?

More Info:

Running busybox inside the cluster and using traceroute I see that it stops at 10.0.2.2. Now I just need to figure out what 10.0.2.2 is and why it stops there. ;)  I don't understand the networking aspects of getting from inside K8S to a internet resource when K8S is running on a VM from minikube.  

```
/ # traceroute 23.99.34.75
traceroute to 23.99.34.75 (23.99.34.75), 30 hops max, 46 byte packets
 1  172.17.0.1 (172.17.0.1)  0.004 ms  0.004 ms  0.001 ms
 2  10.0.2.2 (10.0.2.2)  0.179 ms  0.217 ms  0.183 ms
 3  *  *  *
 4^C
/ #
```

Original Question:

Using Minikube v 6 on OSX. I have an already containerized app that talks to a database on Azure. I thought I would just be able to spin up the container inside K8S and have it work, but DNS inside K8S would only have internal K8S resources, and even if I use the external IP it still seems that egress/ingress is causing problems because the app cannot authenticate to the external database. Can someone point me in the right direction?

I tried "[Services without Selectors](http://kubernetes.io/docs/user-guide/services/)" but no joy...

**Comment by apelisse** [comment]
Please re-post your question to [stackoverflow]
(http://stackoverflow.com/questions/tagged/kubernetes).

We are trying to consolidate the channels to which questions for help/support
are posted so that we can improve our efficiency in responding to your requests,
and to make it easier for you to find answers to frequently asked questions and
how to address common use cases.

We regularly see messages posted in multiple forums, with the full response
thread only in one place or, worse, spread across multiple forums. Also, the
large volume of support issues on github is making it difficult for us to use
issues to identify real bugs.

The Kubernetes team scans stackoverflow on a regular basis, and will try to
ensure your questions don't go unanswered.

Before posting a new question, please search stackoverflow for answers to
similar questions, and also familiarize yourself with:

  * [user guide](http://kubernetes.io/docs/user-guide/)
  * [troubleshooting guide](http://kubernetes.io/docs/admin/cluster-troubleshooting/)

Again, thanks for using Kubernetes.

The Kubernetes Team

**Comment by dstroot** [comment]
Got it!  Thanks.

**Comment by sirvon** [comment]
@dstroot did you make progress on this....whats/where is the stackoverflow post?

**Comment by dstroot** [comment]
@sirvon - it was "user education" I suppose.  I discovered and spun up the explorer container (https://github.com/kubernetes/kubernetes/tree/master/examples/explorer) so I could see how DNS was working inside the cluster.  It turned out DNS was working great.  The true issue was that I did not understand how ENV variables work.  I did not realize that numbers and booleans needed to be in quotes as they do not need to be in quotes elsewhere.  That was my real issue. 

```yaml
    spec:
      containers:
      - name: myname
        image: dstroot/myimage
        env:
        - name: MSSQL_PORT
          value: "1433"    #<- Quoted!
        ports:
        - containerPort: 8000    #<- Unquoted
```
